### PR TITLE
[Rev 0 Demo] Migrate to PrimeReact

### LINKED ISSUE #285: [Rev 0 Demo] Migrate To PrimeReact
Based off of https://github.com/arkinmodi/project-sayyara/issues/270 we should migrate all UI components from Blueprintjs to PrimeReact.

### PR DESCRIPTION
**Pull request by arkinmodi**
Close #285

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 import type { Session } from "next-auth";
 import { SessionProvider } from "next-auth/react";
 import type { AppProps } from "next/app";
 import { FC } from "react";
 import { Provider } from "react-redux";
-import { wrapper } from "../redux/store";
+
+import "primeicons/primeicons.css"; // icons
+import "primereact/resources/primereact.min.css"; // core css
+import "primereact/resources/themes/lara-light-indigo/theme.css"; // theme
```
Comment (arkinmodi) [inline src/pages/_app.tsx:10]: There a few different theme we can pick from. I think the one I picked is the default.
https://primefaces.org/primereact/theming/

Comment (TimChoy) [reply src/pages/_app.tsx:10]: Aren't themes paid, or are there free tiers

Comment (arkinmodi) [reply src/pages/_app.tsx:10]: All of these are free

```
primereact/resources/themes/bootstrap4-light-blue/theme.css
primereact/resources/themes/bootstrap4-light-purple/theme.css
primereact/resources/themes/bootstrap4-dark-blue/theme.css
primereact/resources/themes/bootstrap4-dark-purple/theme.css
primereact/resources/themes/md-light-indigo/theme.css
primereact/resources/themes/md-light-deeppurple/theme.css
primereact/resources/themes/md-dark-indigo/theme.css
primereact/resources/themes/md-dark-deeppurple/theme.css
primereact/resources/themes/mdc-light-indigo/theme.css
primereact/resources/themes/mdc-light-deeppurple/theme.css
primereact/resources/themes/mdc-dark-indigo/theme.css
primereact/resources/themes/mdc-dark-deeppurple/theme.css
primereact/resources/themes/fluent-light/theme.css
primereact/resources/themes/lara-light-blue/theme.css
primereact/resources/themes/lara-light-indigo/theme.css
primereact/resources/themes/lara-light-purple/theme.css
primereact/resources/themes/lara-light-teal/theme.css
primereact/resources/themes/lara-dark-blue/theme.css
primereact/resources/themes/lara-dark-indigo/theme.css
primereact/resources/themes/lara-dark-purple/theme.css
primereact/resources/themes/lara-dark-teal/theme.css
primereact/resources/themes/saga-blue/theme.css
primereact/resources/themes/saga-green/theme.css
primereact/resources/themes/saga-orange/theme.css
primereact/resources/themes/saga-purple/theme.css
primereact/resources/themes/vela-blue/theme.css
primereact/resources/themes/vela-green/theme.css
primereact/resources/themes/vela-orange/theme.css
primereact/resources/themes/vela-purple/theme.css
primereact/resources/themes/arya-blue/theme.css
primereact/resources/themes/arya-green/theme.css
primereact/resources/themes/arya-orange/theme.css
primereact/resources/themes/arya-purple/theme.css
primereact/resources/themes/nova/theme.css
primereact/resources/themes/nova-alt/theme.css
primereact/resources/themes/nova-accent/theme.css
primereact/resources/themes/luna-amber/theme.css
primereact/resources/themes/luna-blue/theme.css
primereact/resources/themes/luna-green/theme.css
primereact/resources/themes/luna-pink/theme.css
primereact/resources/themes/rhea/theme.css
```

Comment (TimChoy) [reply src/pages/_app.tsx:10]: I'm assuming its simple to switch between themes? Especially early on? We can use the default for now, and if Nabeel likes another theme we can switch over

Comment (arkinmodi) [reply src/pages/_app.tsx:10]: Should be just changing that import one line

**Review by TimChoy — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -62,7 +59,7 @@
     "prettier-plugin-organize-imports": "^3.2.0",
     "prisma": "^4.4.0",
     "ts-jest": "^29.0.3",
-    "typescript": "4.8.4"
+    "typescript": "4.9.3"
```
Comment (arkinmodi) [inline package.json:62]: I also went ahead and upgraded `typescript` and `next` to latest minor versions. We can use the new TS features.

**Pull request merged**